framework/src/base/MooseApp.C

```diff
@@ -1572,6 +1572,35 @@ MooseApp::getMeshGeneratorOutput(const std::string & name)
   return outputs.back();
 }
 
+std::string
+MooseApp::getCurrentFinalMeshGenerator() const
+{
+  DependencyResolver<std::shared_ptr<MeshGenerator>> resolver;
+
+  // Add all of the dependencies into the resolver and sort them
+  for (const auto & it : _mesh_generators)
+  {
+    // Make sure an item with no dependencies comes out too!
+    resolver.addItem(it.second);
+
+    std::vector<std::string> & generators = it.second->getDependencies();
+    for (const auto & depend_name : generators)
+    {
+      auto depend_it = _mesh_generators.find(depend_name);
+      if (depend_it != _mesh_generators.end())
+        resolver.insertDependency(it.second, depend_it->second);
+    }
+  }
+
+  const auto & ordered_generators = resolver.getSortedValuesSets();
+
+  std::string final_generator_name;
+  if (ordered_generators.size())
+    final_generator_name = ordered_generators.back().back()->name();
```

Comment:
Rather than duplicate this logic, perhaps you could factor out this piece from the execute method and just call it from there? However, see my other comment below as I don't think you should expose this API at all. Especially if it's not always correct.